modules_core/com.smf.jobs.defaults/src/com/smf/jobs/defaults/invoices/CreateFromOrder.java

```diff
@@ -21,109 +23,120 @@
+    var result = new ActionResult();
+    result.setType(Result.Type.SUCCESS);
 
+    try {
+      if (parameters.has("_buttonValue") && "REFRESH".equals(parameters.getString("_buttonValue"))) {
         result.setResponseActionsBuilder(
-                getResponseBuilder()
+            getResponseBuilder()
                 .retryExecution()
                 .refreshGridParameter(ORDER_GRID_PARAM)
-                .showMsgInProcessView(ResponseActionsBuilder.MessageType.SUCCESS, result.getMessage())
         );
-
         return result;
-    }
+      }
 
-    private void initSelection(List<String> selection) {
-        String deselectAllQueryString = "update Order set selected = false where selected = true";
-        OBDal.getInstance().getSession().createQuery(deselectAllQueryString).executeUpdate();
+      var input = getInputContents(getInputClass());
 
-        String selectQueryString = "update Order set selected = true where id in :selection";
-        var selectQuery = OBDal.getInstance().getSession().createQuery(selectQueryString);
-        selectQuery.setParameter("selection", selection);
-        selectQuery.executeUpdate();
-    }
+      if (input.isEmpty()) {
+        var gridSelection = parameters.getJSONObject(ORDER_GRID_PARAM).getJSONArray("_selection");
+        for (int i = 0; i < gridSelection.length(); i++) {
+          JSONObject gridRecord = gridSelection.getJSONObject(i);
+          input.add(OBDal.getInstance().get(Order.class, gridRecord.getString("id")));
+        }
+      }
 
-    private void resetSelection(List<String> selection) {
-        String resetSelectionQueryString = "update Order set selected = false where id in :selection";
-        var resetSelectionQuery = OBDal.getInstance().getSession().createQuery(resetSelectionQueryString);
-        resetSelectionQuery.setParameter("selection", selection);
-        resetSelectionQuery.executeUpdate();
-    }
+      if (!input.isEmpty()) {
+        var selectedIds = input.stream().map(Order::getId).collect(Collectors.toList());
+        initSelection(selectedIds);
 
-    @Override
-    protected Class<Order> getInputClass() {
-        return Order.class;
+        String invoiceDateStr = parameters.getString("invoiceDate");
+        Date invoiceDate;
+
+        if (StringUtils.equals("null", invoiceDateStr)) {
+          invoiceDate = new Date();
+        } else {
+          invoiceDate = JsonUtils.createDateFormat().parse(invoiceDateStr);
+        }
+
+        final Map<String, Object> processParameters = new HashMap<>();
+        processParameters.put("Selection",
+            "Y"); // Selection is used to collect the orders marked by the initSelection method. See old GenerateInvoicesmanual.java
+        processParameters.put("DateInvoiced", invoiceDate);
+
+        var invoiceCreateProcess = OBDal.getInstance().get(Process.class, "134");
+        final var processInstance = CallProcess.getInstance().callProcess(invoiceCreateProcess, null, processParameters,
```

Comment:
You could use the new ProcessInvoices() actions in com.smf.jobs.defaults

Reply:
is out of the scope, this change is produced by code formatting